src/ripple/ripple.js

```diff
@@ -109,25 +110,52 @@
       }
       this.setRippleXY(x, y);
       this.setRippleStyles(true);
-      requestAnimationFrame(this.animFrameHandler.bind(this));
+
+      if (this.timeout) {
+        window.clearTimeout(this.timeout);
+      }
+
+      // We've just clicked, clear past upHandler_ call record
+      this.rippleElement_.mouseUp = false;
+      // Indicate to upHandler_ that ripple hasn't run for this.Constant_.DURATION
+      this.rippleElement_.animationCompleted = false;
+
+      window.requestAnimationFrame(function() {
+        this.animFrameHandler();
+
+        // Allow the animation to complete, even for tap events, which may
+        // trigger multiple mouseup with final event.detail == 1
+        // Ex: mac trackpad tap triggers 2 consecutive event.detail == 0 events
+        // followed by an event.detail=1 mouseup event
+        // we set the animation timeout inside the frame request to
+        // prevent animation flashing during long ui-blocking operations
+        this.timeout = window.setTimeout(function() {
```

Comment:
Absolutely, to confirm, I SHOULD have merged into mdl-1.1, they use window.setTimeout. Could you tell me, in your/colleagues experience, is setTimeout / requestAnimationFrame better accepted and therefore more readable that window.*? I would think we want to give the full context (i.e to which object this method belongs)? Thank you.